sleap/gui/commands.py

```
@@ -3470,13 +3481,203 @@ def do_action(context: CommandContext, params: dict):
+                    return
+
+                # Read and process frame
+                try:
+                    frame = video.get_frame(frame_idx)
+                    if frame.ndim == 2:  # Grayscale frames
+                        frame = cv2.cvtColor(frame, cv2.COLOR_GRAY2BGR)
+                    writer.add_frame(frame, bgr=True)
+                except Exception as e:
+                    writer.close()
+                    os.remove(params["filename"])
+                    raise RuntimeError(f"Failed to write frame {frame_idx}: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)